src/domain-functions.ts

```diff
@@ -49,40 +49,11 @@ function all<Fns extends DomainFunction[]>(
 function collect<Fns extends Record<string, DomainFunction>>(
   fns: Fns,
 ): DomainFunction<UnpackDFObject<Fns>> {
-  return async (input, environment) => {
-    const results = await Promise.all(
-      Object.entries(fns).map(
-        async ([key, fn]) =>
-          [key, await (fn as DomainFunction)(input, environment)] as const,
-      ),
-    )
+  const dfsWithKey = Object.entries(fns).map(([key, df]) =>
+    map(df, (result) => Object.fromEntries([[key, result]])),
+  )
+  return map(all(...dfsWithKey), mergeObjects) as DomainFunction<UnpackDFObject<Fns>>
```

Comment:
Maybe `merge` was a mistake, but we can still...
```suggestion
  return merge(...dfsWithKey) as DomainFunction<UnpackDFObject<Fns>>
```

Reply:
true, but the merge definition is more complex (and rewriting that would bring little benefit). I rather leave the map + all since it is a good example of using simpler combinators to build more complex functionality